In [1]:
import scipy.stats as sps
import numpy as np

### Урок 6

#### 1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):<br>zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],<br>ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].<br>Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy. Полученные значения должны быть равны.<br>Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,а затем с использованием функций из библиотек numpy и pandas.

In [2]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
zp

array([ 35,  45, 190, 200,  40,  70,  54, 150, 120, 110])

In [3]:
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])
ks

array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [4]:
c = np.mean(zp * ks) - np.mean(zp) * np.mean(ks)
c

9157.839999999997

In [5]:
cov = np.cov(ks, zp, ddof=0)
cov

array([[30468.89,  9157.84],
       [ 9157.84,  3494.64]])

In [6]:
r = c / (np.std(ks) * np.std(zp))
r

0.8874900920739158

In [7]:
r = np.corrcoef(ks, zp)
r

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

#### 2. Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:<br>131, 125, 115, 122, 131, 115, 107, 99, 125, 111.<br>Известно, что в генеральной совокупности IQ распределен нормально.<br>Найдите доверительный интервал для математического ожидания с надежностью 0.95.

Границы доверительного интервала находим по формуле:
$$T_{1,2} = \overline{X} \pm t_{\frac{\alpha}{2}}\cdot \frac{\sigma}{\sqrt{n}}$$

Находим $\overline{X}$ по выборке:

In [8]:
a = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
x = np.mean(a)
x

118.1

Находим $t_{\frac{\alpha}{2}}$:

In [9]:
t = sps.t.ppf(0.975, 9)
t

2.2621571627409915

Находим $\sigma$:

In [10]:
s = np.std(a, ddof=1)
s

10.54566788359614

Находим границы доверительного интервала $T_1$ и $T_2$:

In [11]:
t_1 = x - t * (s / np.sqrt(len(a)))
t_1

110.55608365158724

In [12]:
t_2 = x + t * (s / np.sqrt(len(a)))
t_2

125.64391634841274

$$T_{1,2} = [110.556;125.6439]$$ С доверительной вероятности 0,95

#### 3. Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

Границы доверительного интервала находим по формуле:
$$T_{1,2} = \overline{X} \pm Z_{\frac{\alpha}{2}}\cdot \frac{\sigma}{\sqrt{n}}$$
Находим $Z_{\frac{\alpha}{2}}$:

In [13]:
z = sps.norm.ppf(0.975)
z

1.959963984540054

По условию задачи $\overline{X}=174.2, \ \sigma=5, \ n=27$<br>

In [14]:
x1 = 174.2
s1 = 5
n = 27

Находим границы доверительного интервала $T_1$ и $T_2$:

In [15]:
t1 = x1 - z * (s1 / n ** 0.5)
t1

172.3140237765397

In [16]:
t2 = x1 + z * (s1 / n ** 0.5)
t2

176.08597622346028

$$T_{1,2} = [172.3;176.1]$$ С доверительной вероятности 0,95